clash-ghc/src-ghc/Clash/GHC/ClashFlags.hs

```diff
@@ -1,327 +1,328 @@
-  , defFlag "fclash-edalize"                     $ NoArg (liftEwM (setEdalize r))
-  , defFlag "fclash-no-render-enums"             $ NoArg (liftEwM (setNoRenderEnums r))
+ghcClashFlags :: IORef ClashOpts -> [Flag IO]
+ghcClashFlags ref = concatMap (`cfFlags` ref) clashFlags
+
+clashFlags :: [ClashFlag]
+clashFlags =
+  -- TODO: Depending on the type of flag, users need to specify arguments differently. That
+  --       is:
+  --
+  --          * int/word:    -fclash-foo=3
+  --          * str/enum/..: -fclash-foo s
+  --
+  --       Even worse, there's a couple of exceptions to this rule too:
+  --
+  --          * -fclash-debug-history=history.dat
+  --          * -fclash-force-undefined 1
+  --
+  --       We should be liberal in what we accept and allow both types for all flags.
+  --
+  [ debugFlag "debug"
+  , debugInfoFlag "debug-info"
+  , boolFlag "debug-invariants" (opt_debug . dbg_invariants)
+  , boolFlag "count-transformations" (opt_debug . dbg_countTransformations)
+  , debugTransformationsFlag "debug-transformations"
+  , maybeWordFlag "debug-transformations-from" (opt_debug . dbg_transformationsFrom)
+  , maybeWordFlag "debug-transformations-limit" (opt_debug . dbg_transformationsLimit)
+  , maybeStringWithDefaultFlag "debug-history" "history.dat" (opt_debug . dbg_historyFile)
+  , maybeStringFlag "hdldir" opt_hdlDir
+  , hdlSynFlag "hdlsyn"
+  , boolFlag "cache" opt_cachehdl
+  , boolFlag "check-inaccessible-idirs" opt_checkIDir
+  , boolFlag "clear" opt_clear
+  , boolFlag "prim-warn" opt_primWarn
+  , intFlag  "spec-limit" opt_specLimit
+  , intFlag  "inline-limit" opt_inlineLimit
+  , wordFlag "inline-function-limit" opt_inlineFunctionLimit
+  , wordFlag "inline-constant-limit" opt_inlineConstantLimit
+  , wordFlag "evaluator-fuel-limit" opt_evaluatorFuelLimit
+  , intWidthFlag "intwidth"
+  , boolFlag "error-extra" opt_errorExtra
+  , maybeStringFlag "component-prefix" (opt_componentPrefix . packUnpackLens)
+  , boolFlag "old-inline-strategy" (opt_newInlineStrat . iso' not not)
+  , boolFlag "escaped-identifiers" opt_escapedIds
+  , boolFlag "lower-case-basic-identifiers" lowerCaseLens
+  , boolFlag "compile-ultra" opt_ultra
+  , forceUndefinedFlag "fclash-force-undefined"
+  , aggressiveXOptFlag "aggressive-x-optimization"
+  , boolFlag "aggressive-x-optimization-blackboxes" opt_aggressiveXOptBB
+  , wordFlag "inline-workfree-limit" opt_inlineWFCacheLimit
+  , boolFlag "edalize" opt_edalize
+  , boolFlag "render-enums" opt_renderEnums
+
+  -- Deprecated/removed flags:
+  , deprecated "-fclash-float-support is always enabled from Clash 1.6 and onwards" $
+      boolFlag "float-support" voidLens
+
+  , deprecated "-fclash-clean has been removed" $
+      boolFlag "clean" voidLens
+
+  , deprecated "-fclash-nocache deprecated in favor of -fclash-no-cache" $
+      (boolFlag "nocache" (opt_cachehdl . iso' not not)){ cfRender = const [] }
```

Comment:
I think much like deprecated APIs, when we deprecate a flag it should include a message saying it won't be a flag after version 1.X. Then when that version is released it gets removed and just becomes a normal "unknown flag" error

Reply:
Agreed. Can this go in a separate PR? I don't want to battle out in which version things need to deprecate now :smile:

Reply:
The "default" is current + 2 major, I don't think that's controversial here?

Reply:
Okay, I've set them all to 1.10 now. If this leads to any discussion I'm going to rip it from this PR.